R/zchunk_LA124.heat.R

```diff
@@ -40,70 +40,175 @@ module_energy_LA124.heat_DISABLED <- function(command, ...) {
     L1231.out_EJ_R_elec_F_tech_Yh <- get_data(all_data, "temp-data-inject/L1231.out_EJ_R_elec_F_tech_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Create list of heat regions
+    heat_regionIDs <- A_regions$GCAM_region_ID[ A_regions$heat == 1]
+
+    # Fuel inputs to district heat
+    # Process fuel inputs in all regions; some will have the energy assigned to bld/ind, and others have a district heat sector
+    L1011.en_bal_EJ_R_Si_Fi_Yh  %>%
+      filter(sector == "in_heat") %>%
+      mutate(sector = sub("in_", "", sector)) %>%
+      left_join(enduse_fuel_aggregation, by="fuel") %>%
+      select(-fuel, -electricity, -bld, -industry, -trn) %>%
+      rename(fuel=heat) %>%
```

Comment:
It's not at all clear what's happening in this pipeline.  Comments should give us an idea of what we're starting with and what we will end up with.  Some explanation of why we are doing this would also not be amiss.

Be especially cognizant of this when using the negative form of `select`.  When you select a (positive) list of columns, then I at least know what will be in the resulting table.  When you tell me what columns you are dropping (i.e., by using the negative form), it may not be obvious what is left.